modules/gapi/test/s11n/gapi_sample_pipelines_s11n.cpp

```diff
@@ -484,4 +489,303 @@ TEST(S11N, Pipeline_GArray_GOpaque_Multinode)
+
+    cv::GMat in, out;
+    cv::GArray<cv::gapi::wip::draw::Prim> arr;
+    out = cv::gapi::wip::draw::render3ch(in, arr);
+    cv::GComputation comp(cv::GIn(in, arr), cv::GOut(out));
+
+    auto serialized = cv::gapi::serialize(comp);
+    auto dc = cv::gapi::deserialize<cv::GComputation>(serialized);
+
+    cv::Mat input(1920, 1080, CV_8UC3);
+    cv::randu(input, cv::Scalar::all(0), cv::Scalar::all(255));
+    cv::Mat ref_mat = input.clone();
+    dc.apply(cv::gin(input, prims), cv::gout(input));
+
+    // OpenCV code //////////////////////////////////////////////////////////////
+    for (int i = 0; i < rects_num; ++i) {
+        cv::Rect rect(200 + i, 200 + i, 200, 200);
+        cv::rectangle(ref_mat, rect, color, thick, lt, shift);
+    }
+
+    for (int i = 0; i < rects_num; ++i) {
+        cv::Rect mos(200 + i, 200 + i, 200, 200);
+         drawMosaicRef(ref_mat, mos, cellsz);
+    }
+
+    for (int i = 0; i < text_num; ++i) {
+        cv::Point org(200 + i, 200 + i);
+        cv::putText(ref_mat, text, org, ff, fs, color, thick, lt, blo);
+    }
+
+    for (int i = 0; i < image_num; ++i) {
+        cv::Point org_img = {tl.x + i, tl.y + rect_img.size().height + i};
+        blendImageRef(ref_mat, org_img, img, alpha);
+    }
+
+    cv::circle(ref_mat, center, rad, color, thick, lt, shift);
+    cv::line(ref_mat, center, point_next, color, thick, lt, shift);
+    std::vector<std::vector<cv::Point>> pp{points};
+    cv::fillPoly(ref_mat, pp, color, lt, shift);
+
+    EXPECT_EQ(cv::norm(input,  ref_mat), 0);
+}
+
+#ifdef HAVE_FREETYPE
+namespace
+{
+static std::wstring to_wstring(const char* bytes)
+{
+    std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> converter;
+    return converter.from_bytes(bytes);
+}
+}
+
+TEST(S11N, Pipeline_Render_FText_NV12)
+{
+    std::wstring text = to_wstring("\xe4\xbd\xa0\xe5\xa5\xbd\xef\xbc\x8c\xe4\xb8\x96\xe7\x95\x8c");
+    cv::Point org(100, 100);
+    int fh = 64;
+    cv::Scalar color(200, 100, 25);
+
+    cv::gapi::wip::draw::Prims prims;
+    prims.emplace_back(cv::gapi::wip::draw::FText{text, org, fh, color});
+
+    cv::GMat y_in, uv_in, y_out, uv_out;
+    cv::GArray<cv::gapi::wip::draw::Prim> arr;
+    std::tie(y_out, uv_out) = cv::gapi::wip::draw::renderNV12(y_in, uv_in, arr);
+    cv::GComputation comp(cv::GIn(y_in, uv_in, arr), cv::GOut(y_out, uv_out));
+
+    auto serialized = cv::gapi::serialize(comp);
+    auto dc = cv::gapi::deserialize<cv::GComputation>(serialized);
+
+    cv::Mat y(1920, 1080, CV_8UC1);
+    cv::Mat uv(960, 540, CV_8UC2);
+    cv::randu(y, cv::Scalar::all(0), cv::Scalar::all(255));
+    cv::randu(uv, cv::Scalar::all(0), cv::Scalar::all(255));
+    auto y_ref = y.clone();
+    auto uv_ref = uv.clone();
+
+    EXPECT_NO_THROW(dc.apply(cv::gin(y, uv, prims), cv::gout(y, uv),
+                             cv::compile_args(cv::gapi::wip::draw::freetype_font{
+                             "/usr/share/fonts/truetype/wqy/wqy-microhei.ttc"
```

Comment:
> /usr/share/fonts/truetype/wqy/wqy-microhei.ttc

Absolute paths should be checked (we should have clear error message, especially in tests).
Consider making these paths configured.

Reply:
Configured from where? Didn't quite get you here. We already have exactly the same logic in https://github.com/opencv/opencv/blob/ea4b491a7311d3ea8d04ade12da09b149d998b0f/modules/gapi/test/render/gapi_render_tests_ocv.cpp#L95. Can this be fixed as a separate PR?

Reply:
Yes as I said, please fix it separately. Have a look on inference or streaming tests to see how `OPENCV_TEST_DATA_PATH` is used. As I also said, you need to put a font to `opencv_extra` to make it work this way.